captured-training-yolov5-segmentation/picsellia/docker_run_training_yolov5_segmentation.py

```diff
@@ -29,24 +29,55 @@
 base_imgdir = experiment.png_dir
 
 parameters = experiment.get_log(name='parameters').data
+attached_datasets = experiment.list_attached_dataset_versions()
 
-if len(experiment.list_attached_dataset_versions())==3:
-    train_ds, val_ds, test_ds = experiment.get_dataset(name='train'), experiment.get_dataset(name='val'), experiment.get_dataset(name='test')
+if len(attached_datasets) == 3:
+    attached_names = [dataset.version for dataset in attached_datasets]
+    if "train" not in attached_names:
+        raise ResourceNotFoundError("Found 3 attached datasets, but can't find any 'train' dataset.\n \
+                                    expecting 'train', 'test', ('val' or 'eval')")
+    else:
+        train_ds = experiment.get_dataset(name="train")
     
-    for data_type, dataset in {'train' : train_ds, 'val' : val_ds, 'test' : test_ds}.items():
-        annotation_path = dataset.export_annotation_file(AnnotationFileType.COCO, current_dir)
-        f = open(annotation_path)
-        annotations_dict = json.load(f)
-        annotations_coco=COCO(annotation_path)
-        
-        if data_type=='train':
-            labelmap = {}
-            for x in annotations_dict['categories']:
-                labelmap[str(x['id'])] = x['name']
-        
-        dataset.list_assets().download(target_path=os.path.join(base_imgdir, data_type, 'images'), max_workers=8)
-        picsellia_utils.create_yolo_segmentation_label(experiment, data_type, annotations_dict, annotations_coco)
+    if "test" not in attached_names:
+        raise ResourceNotFoundError("Found 3 attached datasets, but can't find any 'test' dataset.\n \
+                                    expecting 'train', 'test', ('val' or 'eval')")
+    else:
+        test_ds = experiment.get_dataset(name="test")
+    
+    if "val" not in attached_names:
+        if "eval" not in attached_names:
+            raise ResourceNotFoundError("Found 3 attached datasets, but can't find any ('val' or 'eval') dataset.\n \
+                                        expecting 'train', 'test', ('val' or 'eval')")
+        else:
+            val_ds = experiment.get_dataset(name="eval")
+    else:
+        val_ds = experiment.get_dataset(name="val")
     
+    label_names = [label.name for label in train_ds.list_labels()]
+    
+    for data_type, dataset in {
+        "train": train_ds,
+        "val": val_ds,
+        "test": test_ds,
+    }.items():
+        coco_annotation = dataset.build_coco_file_locally(enforced_ordered_categories=label_names)
+        annotations_dict = coco_annotation.dict()
+        annotations_path = "annotations.json"
+        with open(annotations_path, 'w') as f:
+            f.write(json.dumps(annotations_dict))
+        annotations_coco = COCO(annotations_path)
```

Comment:
Est-ce qu'on pourrait faire un truc comme ça pour éviter d'écrire dans un fichier pour le relire derrière ? 
```
class PxlCOCO(COCO):
    def __init__(self, dataset):
        """Replace COCO() constructor with this one, so it will not use a file but use given COCO dataset.

        :param dataset: COCO dataset to use
        """
        assert type(dataset) == dict, "annotation file format {} not supported".format(
            type(dataset)
        )

        self.dataset, self.anns, self.cats, self.imgs = dict(), dict(), dict(), dict()
        self.imgToAnns, self.catToImgs = defaultdict(list), defaultdict(list)
        self.dataset = dataset
        self.createIndex()
```

`annotations_coco = PxlCOCO(coco_annotation.dict())`

Pas sûr que ça marche du premier coup mais ça vaudrait le coup